src/ScheduleFunctions.cpp

```diff
@@ -1193,10 +1193,17 @@ class InjectFunctionRealization : public IRMutator {
             internal_assert(iter != dims.end());
             start_fuse = (int)(iter - dims.begin());
         }
+
+        const auto &env_iter = env.find(fuse_level.func());
+        internal_assert(env_iter != env.end());
+        const auto &parent_func = env_iter->second;
+
+        const vector<Dim> &parent_dims = (fuse_level.stage_index() == 0) ? parent_func.definition().schedule().dims() : parent_func.update(fuse_level.stage_index() - 1).schedule().dims();
+        int fused_vars_num = dims.size() - start_fuse - 1;
```

Comment:
Does the code above guarantee this won't underflow?

Reply:
There is a check above to make sure that start_fuse is in range, however, I moved this line up to make it more visible.